src/uct/base/uct_iface.c

```diff
@@ -732,3 +732,53 @@ uct_ep_keepalive_check(uct_ep_h tl_ep, uct_keepalive_info_t *ka, unsigned flags,
 
     return UCS_OK;
 }
+
+/* Read boot_id GUID or use machine_guid */
+static uint64_t uct_iface_get_system_id()
+{
+    uint64_t high;
+    uint64_t low;
+    ucs_status_t status;
+
+    status = ucs_sys_get_boot_id(&high, &low);
+    if (status == UCS_OK) {
+        return high ^ low;
+    }
+
+    return ucs_machine_guid();
+}
+
+void UCS_F_NOOPTIMIZE /* GCC failed to compile it in release mode */
+uct_iface_get_local_address(uct_iface_ext_local_addr_t *ext_addr)
+{
+    ext_addr->super.id = uct_iface_get_system_id() &
+                         ~UCT_IFACE_LOCAL_ADDR_FLAG_EXT;
+
+    if (!ucs_sys_ns_is_default(UCS_SYS_NS_TYPE_IPC)) {
+        ext_addr->super.id |= UCT_IFACE_LOCAL_ADDR_FLAG_EXT;
+        ext_addr->ipc_ns    = ucs_sys_get_ns(UCS_SYS_NS_TYPE_IPC);
+    }
+}
+
+int uct_iface_local_is_reachable(uct_iface_ext_local_addr_t *ext_addr)
+{
```

Comment:
is this function really relevant for all ifaces? i mean, should it be in this file?